src/components/NewProjectFlow/flow/Search.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { Grid, Box, useTheme } from '@mui/material';
+import { PillListItem, Textfield } from '@terraware/web-components';
+import { PillList } from '@terraware/web-components';
+import strings from 'src/strings';
+import { ProjectEntitiesFilters } from 'src/components/NewProjectFlow/flow/useProjectEntitySelection';
+
+interface SearchProps {
+  searchValue: string;
+  onSearch: (value: string) => void;
+  filters: ProjectEntitiesFilters;
+  setFilters: React.Dispatch<React.SetStateAction<ProjectEntitiesFilters>>;
+}
+
+type PillListItemWithEmptyValue = PillListItem<string> & { emptyValue: unknown };
+
+export default function Search(props: SearchProps): JSX.Element | null {
+  const { searchValue, onSearch, filters, setFilters } = props;
+
+  const theme = useTheme();
+
+  const [filterPillData, setFilterPillData] = useState<PillListItemWithEmptyValue[]>([]);
+
+  useEffect(() => {
```

Comment:
filterPillData could be a useMemo

Reply:
This is updated / changed in https://github.com/terraware/terraware-web/pull/1843